src/lightbox/tp-lightbox.ts

```diff
@@ -0,0 +1,363 @@
+import { TPLightboxNextElement } from './tp-lightbox-next';
+import { TPLightboxTriggerElement } from './tp-lightbox-trigger';
+import { TPLightboxCountElement } from './tp-lightbox-count';
+
+/**
+ * TP Lightbox.
+ */
+export class TPLightboxElement extends HTMLElement {
+	/**
+	 * Properties.
+	 */
+	protected currentTemplate: HTMLTemplateElement | null = null;
+	protected currentGroup: string = '';
+	protected allGroups: NodeListOf<TPLightboxTriggerElement> | null = null;
+
+	/**
+	 * Constructor.
+	 */
+	constructor() {
+		super();
+
+		// Event listeners.
+		this.querySelector( 'dialog' )?.addEventListener( 'click', this.handleDialogClick.bind( this ) );
+	}
+
+	/**
+	 * Get observed attributes.
+	 *
+	 * @return {Array} List of observed attributes.
+	 */
+	static get observedAttributes(): string[] {
+		return [ 'open', 'index', 'total', 'close-on-overlay-click', 'loading' ];
+	}
+
+	/**
+	 * Attribute changed callback.
+	 *
+	 * @param {string} name     Attribute name.
+	 * @param {string} oldValue Old value.
+	 * @param {string} newValue New value.
+	 */
+	attributeChangedCallback( name: string = '', oldValue: string = '', newValue: string = '' ): void {
+		// Prevent redundant updates.
+		if ( oldValue === newValue ) {
+			return;
+		}
+
+		this.dispatchEvent( new CustomEvent( 'change' ) );
+
+		// Trigger current index target if index has changed.
+		if ( 'index' === name ) {
+			this.triggerCurrentIndexTarget();
+		}
+	}
+
+	/**
+	 * Get template.
+	 */
+	get template(): HTMLTemplateElement | null {
+		return this.currentTemplate;
+	}
+
+	/**
+	 * Set template.
+	 *
+	 * @param {HTMLTemplateElement} template The template.
+	 */
+	set template( template: HTMLTemplateElement | null ) {
+		// Set the template.
+		this.currentTemplate = template;
+		this.dispatchEvent( new CustomEvent( 'template-set' ) );
+
+		// Get lightbox content element.
+		const content: TPLightboxContentElement | null = this.querySelector( 'tp-lightbox-content' );
+		if ( ! content ) {
+			return;
+		}
+
+		// Check if we have a template.
+		if ( this.currentTemplate ) {
+			// We do, update content with template's content.
+			// We do this rather than a string to avoid script injection.
+			const templateContent: Node = this.currentTemplate.content.cloneNode( true );
+			content.replaceChildren( templateContent );
+			this.dispatchEvent( new CustomEvent( 'content-change' ) );
+
+			setTimeout( (): void => {
+				this.prepareImageLoading();
+				this.prepareNavigation();
+			}, 0 );
+		} else {
+			// We don't, set content as empty.
+			content.innerHTML = '';
+		}
+	}
+
+	/**
+	 * Get current group.
+	 */
+	get group(): string {
+		return this.currentGroup;
+	}
+
+	/**
+	 * Set current group.
+	 *
+	 * @param {string} group Group name.
+	 */
+	set group( group: string ) {
+		this.currentGroup = group;
+	}
+
+	/**
+	 * Get current index.
+	 */
+	get currentIndex(): number {
+		return parseInt( this.getAttribute( 'index' ) ?? '1' );
```

Comment:
@junaidbhura Let' change this to `return parseInt( this.getAttribute( 'index' ) || '1' );`